src/main/java/org/inferred/freebuilder/processor/util/feature/FunctionPackage.java

```diff
@@ -30,12 +32,18 @@ protected FunctionPackage testDefault() {
 
         @Override
         protected FunctionPackage forEnvironment(ProcessingEnvironment env) {
-          TypeElement element = env.getElementUtils()
-              .getTypeElement(UNARY_OPERATOR.getQualifiedName().toString());
-          return (element != null) ? AVAILABLE : UNAVAILABLE;
+          if (runningInEclipse()) {
+            // Eclipse is bugged: sourceVersion will never be > 7.
+            // Work around this by checking for the presence of java.util.function.Consumer instead.
+            return hasType(env.getElementUtils(), CONSUMER) ? AVAILABLE : UNAVAILABLE;
```

Comment:
why does this not suffer from the problem of compiling in Eclipse 1.8 for Java 1.7? Or is this just a tradeoff that you'll have to make whatever way?